package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,352 @@
+        if n_atoms_now != self.n_atoms :
+            raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                             " atoms. MDAnalysis is unable to deal with variable"
+                             " topology!"%(frame, n_atoms_now, self.n_atoms))
+        else :
+            ts.positions = frame_positions
+
+        # set particle velocities
+        if self.has_velocities:
+            frame_velocities = particle_group['velocity/value'][frame, :]
+            n_atoms_now = frame_velocities.shape[0]
+            if n_atoms_now != self.n_atoms :
+                raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                                 " atoms. MDAnalysis is unable to deal with variable"
+                                 " topology!"%(frame, n_atoms_now, self.n_atoms))
+            else :
+                ts.velocities = frame_velocities
+
+        # set particle forces
+        if self.has_forces:
+            frame_forces = particle_group['force/value'][frame, :]
+            n_atoms_now = frame_forces.shape[0]
+            if n_atoms_now != self.n_atoms :
+                raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                                 " atoms. MDAnalysis is unable to deal with variable"
+                                 " topology!"%(frame, n_atoms_now, self.n_atoms))
+            else :
+                ts.forces = frame_forces
+
+        # unit conversion
+        self._translate_h5md_units()  # fills units dictionary
+        if self.convert_units:
+            #self.ts.time = self.convert_time_to_native(self.ts.time)
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+            self.convert_pos_from_native(self.ts.positions)
+            if self.has_velocities:
+                self.convert_velocities_from_native(self.ts.velocities)
+            if self.has_forces:
+                self.convert_forces_from_native(self.ts.forces)
+
+        return ts
+
+
+    def _read_next_timestep(self) :
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+
+    def _translate_h5md_units(self):
+        """converts units from H5MD to MDAnalysis notation
+        and fills units dictionary"""
+
+        velocity_dict = {
+        'nm ps-1': 'nm/ps',
+        'Angstrom ps-1': 'Angstrom/ps'
+        }
+        force_dict = {
+        'kJ mol-1 nm-1': 'kJ/(mol*nm)',
+        'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)'
+        }
+
+        if 'units' in self._particle_group['position/time'].attrs:
+            try:
+                self.units['time'] = self._particle_group['position/time'].attrs['units']
+            except KeyError:
+                print("Unit %d not recognized by H5MDReader. Please "
+                      "raise an issue on `MDAnalysis github <https://github.com/MDAnalysis/mdanalysis>`_"
+                      ""%(self._particle_group['position/time'].attrs['units']))
```

Comment:
raise a `RuntimeError` – we do not use `print()` anywhere in the library. If it's a problem then you have the following options
- raise an issue
- raise a warning
- log output through a logger

In this case we cannot continue so we need to fail, not just continue with a message that a user might never see (because this code might be used in other code that just never displays print messages to a user, e.g., in a GUI).